plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginCommandHandler.java

```diff
@@ -6,15 +6,18 @@
 import org.eclipse.ui.IWorkbenchWindow;
 import org.eclipse.ui.handlers.HandlerUtil;
 import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.window.SameShellProvider;
 
 public class GoogleLoginCommandHandler extends AbstractHandler {
 
-	public Object execute(ExecutionEvent event) throws ExecutionException {
-	  boolean success = new GoogleLoginTemporaryTester().testLogin();
-
+  public Object execute(ExecutionEvent event) throws ExecutionException {
     IWorkbenchWindow window = HandlerUtil.getActiveWorkbenchWindowChecked(event);
+
+    SameShellProvider shellProvider = new SameShellProvider(window.getShell());
```

Comment:
I think it's better to push this kind of thing onto your test class (e.g., offer `testLogin(Shell)` and `#testLogin(IShellProvider)` with the former just calling the latter with `SameShellSupplier`)